File: easy/1281/1281.py
#!/usr/bin/env python3

class Solution(object):
    def subtractProductAndSum(self, n):
        """
        :type n: int
        :rtype: int
        """
        n_sum = 0
        n_product = 1
        
        while n != 0:
            curr = n % 10
            print(curr)
            n_sum = n_sum + curr
            n_product = n_product * curr
            n = n / 10
        
        return n_product - n_sum

File: easy/1281/1281.cpp
#include <string>
#include <vector>

class Solution {
public:
    int subtractProductAndSum(int n) {
        int len = std::to_string(n).size();
        int sum=0 , product=1, i, curr;
        
        for(i = 0; i < len; i++){
            curr = n % 10;            
            sum = sum + curr;
            product = product * curr;
            n = n / 10;
        }
        return product - sum;
    }
};

int main(){
    std::vector<int> test_cases = {234, 4421};
    int size=test_cases.size(), i, curr;

    Solution s;

    for(i = 0; i < size; i++){
        curr = test_cases[i];
        printf("test case - %d :: answer - %d\n",curr,s.subtractProductAndSum(curr));
    }

    return 0;
}